homeassistant/components/sma/sensor.py

```diff
@@ -164,21 +153,61 @@ async def async_sma(event):
         for sensor in hass_sensors:
             sensor.async_update_values()
 
-    interval = config.get(CONF_SCAN_INTERVAL) or timedelta(seconds=5)
-    async_track_time_interval(hass, async_sma, interval)
+    interval = config_entry.options.get(CONF_SCAN_INTERVAL) or timedelta(
+        seconds=DEFAULT_SCAN_INTERVAL
+    )
 
+    coordinator = DataUpdateCoordinator(
```

Comment:
Usually we setup the coordinator in `__init__.py` and store it in `hass.data[DOMAIN][entry.entry_id]`

https://developers.home-assistant.io/docs/integration_fetching_data/#coordinated-single-api-poll-for-data-for-all-entities

Also if you do that you can call `await coordinator.async_config_entry_first_refresh()` which will automatically retry setup later if the device is offline

Reply:
Thanks, I see that is also the way it is done in `powerwall`. I will try to move some code to `__init__.py`

Reply:
Resolved with ccae47d